### PR TITLE
✨ Add QC Metrics Data to Aligned Reads Files (#506)

### PR DESCRIPTION
**Pull request by mistryrn**
Fetch corresponding Aligned Reads QC file data from RDPC gateway to add `metrics` data to Aligned Reads files

Commits:
* 📦️ Add `graphql-request` Library
* 🔧 Add Data Center Gateway URL to Config
* ✨ Add QC Metrics Data to Aligned Reads Files
* 🛂 Create Ego Score Proxy

### CODE REVIEW

File: src/external/dataCenterGateway.ts [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+  if (!runId || !runId.length) {
+    logger.error('runId is required');
+    throw new Error('runId is required');
+  }
+
+  const config = await getAppConfig();
+  const url = config.datacenter.gatewayUrl;
+  const graphQLClient = new GraphQLClient(url, {
+    headers: {
+      authorization: `Bearer ${config.auth.jwtKey}`,
```
Comment (mistryrn) [inline src/external/dataCenterGateway.ts:38]: @joneubank one thing I was unsure of here was whether or not this `config.auth.jwtKey` would have the correct auth for RDPC. My local environment has auth turned off for files-service, so when I was testing locally I just hardcoded my own jwt in.

File: src/config.ts [hunk 2]
```diff
@@ -166,6 +170,11 @@ const buildAppConfig = async (secrets: any): Promise<AppConfig> => {
       policy,
       writeScope: `${policy}.WRITE`,
       readScope: `${policy}.READ`,
+      egoRootRest: process.env.EGO_ROOT_REST || 'http://localhost:8081',
+      egoScoreProxyClientId:
```
Comment (mistryrn) [inline src/config.ts:174]: I followed the same pattern as a few other Vault-protected secrets for this. Haven't tested the Vault stuff yet, just been setting these in my .env

File: src/services/ego.ts [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+  if (authResponse.error) {
+    throw new Error(
+      `Failed to authorize application: ${(authResponse as EgoAccessTokenError).error_description}`,
+    );
+  }
+
+  return (authResponse as EgoAccessToken).access_token;
+};
+
+export const createScoreAuthClient = async () => {
```
Comment (mistryrn) [inline src/services/ego.ts:93]: I copied only the necessary pieces from https://github.com/icgc-argo/platform-api/blob/develop/src/services/ego/scoreAuthClient.ts and https://github.com/icgc-argo/platform-api/blob/develop/src/services/ego/index.ts for this

File: .env.example [hunk 4]
```diff
@@ -81,6 +84,7 @@ DC_REGISTRY_URL=https://dc.dev.argo.cancercollaboratory.org/
 DC_ID=song.collab
 DC_URL=https://song.rdpc.cancercollaboratory.org/
 DC_BATCH_SIZE=50
+DC_GATEWAY_URL=https://api.rdpc.cancercollaboratory.org/graphql
```
Comment (mistryrn) [inline .env.example:87]: Followed the same pattern as for the `DC_URL` variable in the .env.example here, giving the example of the prod endpoint.

Comment (joneubank) [reply .env.example:87]: lets change this to the QA URL as the provided default for devs: https://api.rdpc-qa.cancercollaboratory.org/graphql

File: src/config.ts [hunk 5]
```diff
@@ -54,6 +54,9 @@ export interface AppConfig {
     policy: string;
     writeScope: string;
     readScope: string;
+    egoRootRest: string;
+    egoScoreProxyClientId: string;
+    egoScoreProxyClientSecret: string;
```
Comment (joneubank) [inline src/config.ts:59]: lets drop the `scoreProxy` part of the name, now that we know it doesnt use the score-proxy permissions.

File: src/config.ts (hunk 2 above)
Comment (joneubank) [reply src/config.ts:174]: same change for the ENV file variables as well...

File: src/services/ego.ts (hunk 3 above)
Comment (joneubank) [reply src/services/ego.ts:93]: for naming consistency, we aren't getting auth for Score here, just getting Ego auth in general. lets rename to `createAuthClient`

**Review by joneubank — CHANGES_REQUESTED** [review]
Looks great, thanks for the updates. Two requests for change:
1 - naming of auth stuff we don't want the name "Score" since that isnt what we need auth for. Let's keep it generic as suggested in comments
2 - GQL Query file location - awesome job keeping this separate from the JS logic. Can you make a directory for `external/dataCenterGateway` and then move the `/gql` directory inside there like `external/dataCenterGateway/gql/QUERY_EXAMPLE.gql`, since that is the service the gql belongs to. we can then move `dataCenterGateway.ts` to `external/dataCenterGateway/index.ts` and it won't even impact your imports.

**Comment by mistryrn** [comment]
Thanks for the feedback, @joneubank ! All really good ideas. I have renamed things to be more generic, and refactored to move the data center gateway code into its own directory within src/external

**Review by joneubank — APPROVED** [review]

**Pull request merged**